src/pages/home/HomePage.tsx

```diff
@@ -1,11 +1,15 @@
 import { useGetEventsQuery } from '@/entities/event/api/eventApi';
 import {DateSlider} from '@/features/calendarFilter';
 import {HomePageTitle} from '@/features/townFilter';
+import { useAppSelector } from '@/shared/model';
 import { EventsList } from '@/widgets/EventsList';
-import {FC} from 'react';
+import { ReactElement } from 'react';
 
-const HomePage: FC = () => {
-  const { data: events = {results: []}, isLoading } = useGetEventsQuery();
+export function HomePage(): ReactElement {
+  const searchFilter = useAppSelector(state => state.globalFilter.search);
+  const { data: events = {results: []}, isLoading, isError } = useGetEventsQuery(searchFilter && searchFilter);
```

Comment:
Why can't we use simple `useGetEventsQuery(searchFilter)`  here?

Reply:
Or, if you apply my suggestion above, this call would be replaced with `useGetEventsQuery({search: searchFilter})`

Reply:
In this case, when we add new filters, this call will remain the same, unchanged